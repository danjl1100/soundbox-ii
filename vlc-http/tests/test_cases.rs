// Copyright (C) 2021-2024  Daniel Lambert. Licensed under GPL-3.0-or-later, see /COPYING file for details

//! Test case runner for `vlc_http`

// yes panic, it's tests!
#![allow(clippy::panic)]

use test_log::test;
use tracing::{trace_span, warn_span};

mod common;

#[test]
fn test_cases() {
    insta::glob!("inputs/*.txt", test_case);
}

fn test_case(input: &std::path::Path) {
    let name = input
        .file_name()
        .expect("input file has a name")
        .to_str()
        .expect("test input filenames are UTF-8");
    let span = warn_span!("test_case", %name);
    let _guard = span.enter();

    let span = trace_span!("test_case", input_path=%input.display());
    let _guard = span.enter();

    let input = std::fs::read_to_string(input).expect("test input file exists");
    let output = common::run_input(&input);
    insta::assert_ron_snapshot!(output);
}
