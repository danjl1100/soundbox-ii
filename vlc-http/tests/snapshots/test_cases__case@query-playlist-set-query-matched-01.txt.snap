---
source: vlc-http/tests/test_cases.rs
expression: output
input_file: vlc-http/tests/inputs/query-playlist-set-query-matched-01.txt
---
[
  Harness(Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
    },
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json?command=pl_play&id=1",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
    },
    current_item_id: Some((1, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
    },
    current_item_id: Some((1, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
    },
    current_item_id: Some((1, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json?command=in_enqueue&input=file:///new1",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
      2: "file:///new1",
    },
    current_item_id: Some((1, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json?command=in_enqueue&input=file:///new2",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
    },
    current_item_id: Some((1, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json?command=in_enqueue&input=file:///new3",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((1, Playing)),
  )),
  Output([
    {
      "duration_secs": 214,
      "id": 2,
      "name": "Item 2",
      "url": "file:///new1",
    },
    {
      "duration_secs": 300,
      "id": 3,
      "name": "Item 3",
      "url": "file:///new2",
    },
    {
      "duration_secs": 407,
      "id": 4,
      "name": "Item 4",
      "url": "file:///new3",
    },
  ]),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json?command=pl_play&id=2",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((2, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((2, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((2, Playing)),
  )),
  Output([
    {
      "duration_secs": 300,
      "id": 3,
      "name": "Item 3",
      "url": "file:///new2",
    },
    {
      "duration_secs": 407,
      "id": 4,
      "name": "Item 4",
      "url": "file:///new3",
    },
  ]),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json?command=pl_play&id=3",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((3, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((3, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json",
  ), Model(
    items: {
      0: "file:///existing1",
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((3, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json?command=pl_delete&id=0",
  ), Model(
    items: {
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((3, Playing)),
  )),
  Output([
    {
      "duration_secs": 407,
      "id": 4,
      "name": "Item 4",
      "url": "file:///new3",
    },
  ]),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json?command=pl_play&id=4",
  ), Model(
    items: {
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((4, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/status.json",
  ), Model(
    items: {
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((4, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json",
  ), Model(
    items: {
      1: "file:///existing2",
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((4, Playing)),
  )),
  LogEntry(Endpoint(
    path_and_query: "/requests/playlist.json?command=pl_delete&id=1",
  ), Model(
    items: {
      2: "file:///new1",
      3: "file:///new2",
      4: "file:///new3",
    },
    current_item_id: Some((4, Playing)),
  )),
  Output([]),
]
